src/caret_analyze/architecture/struct/node.py

```diff
@@ -163,3 +164,25 @@ def to_value(self) -> NodeStructValue:
             else tuple(v.to_value() for v in self.callback_groups),
             None if self.variable_passings is None
             else tuple(v.to_value() for v in self.variable_passings))
+
+    def assign_message_context(self, context_type, sub_topic_name: str, pub_topic_name: str):
+        path = self.get_path(sub_topic_name, pub_topic_name)
+        path.message_context =\
+            MessageContextStruct.create_instance(context_type, Dict(),
+                                                 self.node_name, path.subscription,
+                                                 path.publisher, None)
```

Comment:
ItemNotFoundError has occurred here.
I investigate the cause.

Reply:
Resolve it in [d28b3db](https://github.com/tier4/CARET_analyze/pull/196/commits/d28b3db1e323c231279dd1a53007e8d8f58253bb).
If `ItemNotFoundError` has occurred, new `MessageContextStruct` is created and appended.

https://github.com/tier4/CARET_analyze/blob/bb4ae6b5b7cbef83ce8960979df11e53e4a00481/src/caret_analyze/architecture/struct/node.py#L197-L208